### PR TITLE
Fix inconsistent role name 'exposure' to 'exposures' in Causal_Games.…

### LINKED ISSUE #2642: [BUG] Causal Games notebook throws errors
The Causal Games notebook (https://github.com/pgmpy/pgmpy/blob/dev/examples/Causal_Games.ipynb) throws errors because of a change in the role names for DAG. Should be fixed.

### PR DESCRIPTION
**Pull request by aditya29f70**
…ipynb

# DO NOT REMOVE THIS CHECKLIST 

### Your checklist for this pull request
Your PR **won't** get reviewed if this checklist is not present in the PR.

Please complete the following checklist after creating the PR.

- [x] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side) against our **dev branch** (left side). Please do not request your **dev branch**.
- [x] Have you followed all the steps from our [Contributing Guide](https://github.com/pgmpy/pgmpy/blob/dev/Contributing.md)?
- [x] Are all the GitHub Actions checks passing? If not, they will need to be fixed before review. You can reference logs for the failing check to identify the issue.

Did you use a Large language model (LLM) to assist you in making changes? If yes, please go through the following checklist too: 
- [ ] Please include a short paragraph (**not bullet points**) describing the changes you have made. This should at least include the problem your PR solves and the implemented solution. This doesn't need to be a polished description, but it **must** be written **manually** by you without any assistance from any AI tools.
- [ ] Have you **manually** verified that the changes are doing exactly what is expected? Have you considered edge cases?
- [ ] Has the LLM added try-except blocks? They will need to be removed; any error handling must be explicit.
- [ ] If you used LLM for generating tests, they usually need to be compressed into a smaller set of tests. Please **manually** describe what scenarios the tests are testing for.

### Issue number(s) that this pull request fixes
- Fixes #2642

### List of changes to the codebase in this pull request
- Causal_Games.ipynb, the roles were defined using the singular form ("exposure" and "outcome"). We standardize on the plural form when defining roles in the DAG, the example works consistently with the rest of the implementation.

### CODE REVIEW

File: examples/Causal_Games.ipynb [hunk 1]
```diff
@@ -16,7 +16,7 @@
   },
   {
    "cell_type": "code",
-   "execution_count": 1,
+   "execution_count": 17,
```
Comment (DARHWOLF) [inline examples/Causal_Games.ipynb]: why do we need to change this?

File: examples/Causal_Games.ipynb [hunk 2]
```diff
@@ -64,13 +64,13 @@
...
...So4eqY5GLpdNm6dauna1O8oT5ogFBSB177Mffs2WNGbjZq1Mj8Ha2J30lLWYECBVwfF6FHMyvBoQIaIGfPnjWf6dKl8yygNHE7Anrq1Cnjhw4bNkyGDBliZhS5gZ+jFU0JDjVxAyRz5szm062fmFhAKS5g4gJfRYoUMWV7e/fuTdDSbnO0WjQfLCqgAeJM2aPdy0sXCpo46Wwivs8kBtItbsE/zpgxo+ufV7yjJm6AYNqSXiFF4iVAxGQ/BoFRR0vgiXzo5MmTpWbNmq4L7zGTCTaRD1WCQwU0YOgewRT1Yt6SB23VqpX5OxoTM5fxJZUrV3Z93JUrV5rP6tWruz6G4h0V0IAhX7lz5045c+aMK3OUzhO+/GbdunXmU4eJBYv6oAHTsGFD80kPpm0VTgSxdJe0YFEBDRg0FIJAYYEtMLCMObzO4qEEhwqoBTDhHROX/VZsgHpeAk/9+/cP+lKue1RALaBv375GINhzJWj2799vAk/U4qp5GzwqoBZAqd8TTzxhit6ZJB8kbM7EYjFq1KhAr0P5f1RALWHixImmfpbh0cePHw/kGhBKKo/atm0rxYsXD+QalCtRAbUESuqc3ctovuYzlLDzGc3j5FAHDx4c0nMr/4wKqEVQ8N6mTRtTpte5c+eQCSmFEiwKFMfPnz/fdXG94j/6JCxj+PDhZvwJGq1Dhw4pLqScp3379uY8CCdjUhR70Lm4llK/fn0TMGJCAn2d7HLmN5MmTTJfaEz2IY2NjfX9HIo3VEAthp3N+vXrZ/7MbtpscOSH+cmOZkRriRozHgXNScmhYh8qoJazbds2qVatmpldRK0ugsoGS26avPft22citevXrzcF9tTwjh49Wn1Oi1EBDSNtOnToUNP7iRZlQNhDDz0k5cqV+58jU2hHW7FihdlrhRGawBYTEyZMkLJly4bwN1DcoAIaZlCGx47bu3btSvgeO6JR9cOQL2dMCYJMTyea0knjMECMFIqXOUVKaFEBDVMQPvKmc+fOlTVr1hh/EsEkGsskhejoaDMh/sEHHzQF+WhaTZ+EHyqgimIxmgdVFItRAVUUi1EBVRSLUQFVFItRAVUUi1EBVRSLUQFVFItRAVUUi1EBVRSLUQFVFItRAVUUi1EBVRSLUQFVFItRAVUUi1EBVRSLUQFVFLGX/wOGMsIDhTEAWAAAAABJRU5ErkJggg==",
```
Comment (DARHWOLF) [inline examples/Causal_Games.ipynb]: i think these image links have been accidentally cut out. 
I do not think this was supposed to change

File: examples/Causal_Games.ipynb [hunk 3]
```diff
@@ -83,14 +83,14 @@
...
-    "           roles={'exposure': 'X', 'outcome': 'Y'})\n",
+    "           roles={'exposures': 'X', 'outcomes': 'Y'})\n",
     "\n",
     "example1.to_daft(node_pos={'X': (0, 0), 'Y': (2, 0), 'A': (1, 0), 'B': (1, -1)}).render()"
    ]
   },
   {
    "cell_type": "code",
-   "execution_count": 3,
+   "execution_count": 19,
```
Comment (DARHWOLF) [inline examples/Causal_Games.ipynb]: same here

File: examples/Causal_Games.ipynb [hunk 4]
```diff
@@ -157,13 +157,13 @@
...
...cuoPh9nI9CFcq5554bSUV/UcD25/MIu7ORFNtvphs8XAoOqH6isiouIaVQgkUBd4AKqHRwCwrDokLACG0a59EfdCox3wkMYQVFTUqeBLW0lO+h0dg6ItWatODnMLHTaT/cwjBujB8ajfNmUi2kfA4lc3wOwhnXuZdRk52drae1UbyAoHCqQaohXsDzwjWhbzYVu1mm9HQz/4wWys4wQfBPU1HxQs6JYn12sx85cqRkAjQSMwHYFYFmb7bRiBo0DS8mN0GWsHXPLrBmzRo9acw/5Qy/NOqNwLHWmHPUkdOPSswlbF9qYueD4leR8EaAKIZnt4QoBoxBIk/HidssAHTDF65HTXdYcKhV9jdAw0qIwvyk24KJS9SYbhU0Z5TtVkmzYcMGrSFmCxOK7i+44AJtXo8CctXMO7pg6LDhM1K5k2UsJ2xjgmK2EbjBDKAjnh3V/M7z4q6QmC8MDH2UrVq10t7RVK1gScPqTLsXGoGJgKCySXOQJm8qvYjUUmHFYydHfM8996Slz1kUaBbgvBmEieAXiiJoXTbNA/QsM4ZxKoRYBLRgIhf/kJUbKMFjZaPqYmvdBPiWb7/9tu7O9/XXX+vkoteOiC3b7ZcE0KZMCoJgCBS7AHTs2FG7gLa2ZQodQxSPsKD5W2gSTGGyUVyS6axZs0YXNIJH7M5Aex9jRmkqVsOWFjry+SxkjB1zzx93xhzXI4pezyLhJcDs2bO9jh07etnZ2SwO+uLnKlWqeLVq1fLq1avn1a5d26tataq3zTbb5P8bXq1bt/ZeeeUVr6Ry//33e40bN95kTEqVKuVVq1bNq1Onjo4dY1i5cmUvKysr/9/k5OR4PXv29BYuXOiVRNatW+cNGTJE51TBsStbtqxXs2ZNr27duvri53Llym3ybypVquRdcMEF3urVq2O/71g16OZWKcxeuk5YqRYvXqxmKysdqxWrG1qWDbYw7TgxOYptJDIBGqrx79nKhV5crBLG098filw0xRpYJxTkozUy1ZQNsgsl1hzFNER+6UDxt1XBBcMiIbfKJnYE6+g9TYpEBdQwjK1jS6phOIwJqGE4jAmoYTiMCahhOIwJqGE4jAmoYTiMCahhOIwJqGE4jAmoYTiMCahhOIwJqGE4jAmoYTiMCahhOIwJqGE4jAmoYTiMCahhiLv8HyTFA5pavsvuAAAAAElFTkSuQmCC",
```
Comment (DARHWOLF) [inline examples/Causal_Games.ipynb]: same here

File: examples/Causal_Games.ipynb [hunk 5]
```diff
@@ -702,7 +709,7 @@
    "version": "0.3.2"
   },
   "kernelspec": {
-   "display_name": "Python 3 (ipykernel)",
+   "display_name": "Python 3",
```
Comment (DARHWOLF) [inline examples/Causal_Games.ipynb:712]: why is this change here required?

File: examples/Causal_Games.ipynb [hunk 6]
```diff
@@ -583,7 +583,7 @@
      "text": [
       "Are there any active backdoor paths? True\n",
       "No. of potential adjustment sets: 15\n",
-      "If so, what are the possible backdoor adjustment sets? ['D', 'E'], ['D', 'B'], ['A', 'D'], ['C', 'D'], ['D', 'B', 'E'], ['A', 'D', 'E'], ['C', 'D', 'E'], ['A', 'D', 'B'], ['C', 'D', 'B'], ['C', 'A', 'D'], ['A', 'D', 'B', 'E'], ['C', 'D', 'B', 'E'], ['C', 'A', 'D', 'E'], ['C', 'A', 'D', 'B'], ['C', 'A', 'D', 'B', 'E']\n",
+      "If so, what are the possible backdoor adjustment sets? ['A', 'D'], ['D', 'E'], ['D', 'B'], ['C', 'D'], ['A', 'D', 'E'], ['A', 'D', 'B'], ['C', 'A', 'D'], ['D', 'B', 'E'], ['C', 'D', 'E'], ['C', 'D', 'B'], ['A', 'D', 'B', 'E'], ['C', 'A', 'D', 'E'], ['C', 'A', 'D', 'B'], ['C', 'D', 'B', 'E'], ['C', 'A', 'D', 'B', 'E']\n",
```
Comment (DARHWOLF) [inline examples/Causal_Games.ipynb]: I do not think this was supposed to change?

**Review by DARHWOLF — CHANGES_REQUESTED** [review]
I think there are quite a few errors in this PR.
As far as I understand, it was only the role names that were supposed to change. 
I see image links have been edited and several other changes that i think were not supposed to be changed. 

Please have a look.

**Comment by aditya29f70** [comment]
Thanks for pointing this out. I totally agree with your observation.

I only intended to change the role names (from singular to plural). The other changes you’re seeing—such as the image links and execution metadata—were not edited manually.

Since this file is a `.ipynb` (Jupyter Notebook), when I updated the role names and re-ran the notebook, some of the outputs (like images) and metadata were automatically regenerated. This can happen due to differences in the kernel, library versions, or operating system, and it's quite common when committing notebook files to GitHub.

That said, I’ve verified that `examples/Causal_Games.ipynb` is working correctly and all the examples run as expected.

**Comment by aditya29f70** [comment]
https://drive.google.com/file/d/1n7U9ZXhhEzTL42aLXCCxkJlePh_doQRo/view?usp=sharing ;; see it is working fine!

**Comment by aditya29f70** [comment]
https://drive.google.com/file/d/1n7U9ZXhhEzTL42aLXCCxkJlePh_doQRo/view?usp=sharing ;; see it is working fine !

**Comment by ankurankan** [comment]
@aditya29f70 Can you open the notebook, do a restart kernel and run all cells? This will reset the random cell numbers that you have in your current PR.

**Review by ankurankan — CHANGES_REQUESTED** [review]
Refer comment.

**Comment by aditya29f70** [comment]
I’ve addressed the requested changes. Could you please review again?

**Review by ankurankan — APPROVED** [review]

File: examples/Causal_Games.ipynb (hunk 5 above)
Comment (aditya29f70) [reply examples/Causal_Games.ipynb:712]: some of the outputs (like images) and metadata were automatically regenerated. This can happen due to differences in the kernel, library versions, or operating system, and it's quite common when committing notebook files to GitHub.
